### PR TITLE
add user setting for preferred date format

### PR DESCRIPTION
**Pull request by darkLord19**
Fixes https://github.com/mattermost/focalboard/issues/550

### CODE REVIEW

**Comment by darkLord19** [comment]
@chenilim I have started working on the feature but facing a slight issue. preferredDateFormat is not being stored in localStorage. Can you have a look and help out. Also, is there a way to attach debugger into typescript code in focalboard codebase?

**Comment by chenilim** [comment]
Thanks @darkLord19! I can't see at a glance what might be wrong from the code, but you can debug using the Chrome debugger - more details in the [Dev Tips & Tricks here](https://www.focalboard.com/contribute/getting-started/dev-tips/#debugging-the-web-app).

**Comment by darkLord19** [comment]
@chenilim not sure why but `preferredDateFormat` is not getting called

**Comment by jespino** [comment]
@darkLord19 I like the PR, but I guess the missing part is applying this format in the rest of the application, right?

**Comment by darkLord19** [comment]
@jespino yes that's the plan but was facing issue with storing preference in localstorage and haven't got the chance to debug further yet

**Review by jespino — APPROVED** [review]
Looks good to me. Thanks @darkLord19! :confetti_ball:

File: webapp/src/components/sidebar/sidebarSettingsMenu.tsx [hunk 1]
```diff
@@ -49,6 +52,11 @@ const SidebarSettingsMenu = React.memo((props: Props) => {
         setRandomIcons(!randomIcons)
     }
 
+    const [preferredDateFormat, setPreferredDateFormat] = useState(UserSettings.preferredDateFormat)
+    const updatePreferredDateFormat = (newDateFormat: string) => {
+        setPreferredDateFormat(newDateFormat)
```
Comment (jespino) [inline webapp/src/components/sidebar/sidebarSettingsMenu.tsx:57]: I think you also need to set the setting in the local storage here. something like `UserSettings.preferredDateFormat = newDateFormat`.

**Review by jespino — CHANGES_REQUESTED** [review]
I found a detail about storing in the value in the local storage.

File: webapp/src/components/sidebar/sidebarSettingsMenu.tsx (hunk 1 above)
Comment (darkLord19) [reply webapp/src/components/sidebar/sidebarSettingsMenu.tsx:58]: Oh yes. thanks. Silly me thought that state dispatch will call call the setter!

**Comment by darkLord19** [comment]
@jespino I have updated date property inside card view to take format from user preference. is there any other place we are using date display? I didn't find but i haven;t explored focalboard as much so confirming

File: webapp/src/components/propertyValueElement.tsx [hunk 2]
```diff
@@ -152,6 +154,7 @@ const PropertyValueElement = (props:Props): JSX.Element => {
             <EditableDayPicker
                 className='octo-propertyvalue'
                 value={value as string}
+                dateFormat={UserSettings.preferredDateFormat}
```
Comment (jespino) [inline webapp/src/components/propertyValueElement.tsx:157]: I tiny problem here is this is not going to be reactive, not a huge deal, but probably something to fix later.

**Review by jespino — APPROVED** [review]
Looks good to me.

**Comment by harshilsharma63** [comment]
@darkLord19 can you make the lint and test fixes?

**Comment by darkLord19** [comment]
@harshilsharma63 updated

**Pull request merged**